decidim-core/lib/decidim/core/test/shared_examples/resource_search_examples.rb

```diff
@@ -3,19 +3,18 @@
 require "spec_helper"
 
 shared_examples_for "a resource search" do |factory_name|
-  context "when no component is passed" do
-    let(:params) { default_params.merge(component: nil) }
-
-    it "raises an error" do
-      expect { subject }.to raise_error(StandardError, "Missing component")
+  describe "component_id" do
+    before do
+      get(
+        request_path,
+        headers: { "HOST" => component.organization.host }
+      )
     end
-  end
 
-  describe "component_id" do
     it "only returns resources from the given component" do
       external_resource = create(factory_name)
 
-      expect(subject).not_to include(external_resource)
+      expect(subject).not_to include(CGI.escapeHTML(translated(external_resource.try(:title) || external_resource.try(:name))))
```

Comment:
I see that these three methods are called a lot. How about changing this to a custom Matcher? I'm afraid that in the future we could change the escaping method to another, and we'd need to change it by search and replace. I was thinking something like "include_escaped_and_translated", but of course if you have a better naming, go ahead. What do you think?